src/visualizers/VisualizerHistogram.ts

```diff
@@ -0,0 +1,211 @@
+    }
+
+    // Create an array with the number of factors of
+    // each element at the corresponding index of the array.
+    factorArray(): number[] {
+        const factorArray = []
+        for (
+            let i = Math.max(this.firstIndex, this.seq.first);
+            i < Math.min(this.terms, this.seq.last);
+            i++
+        ) {
+            let counter = 0
+            const factors = this.seq.getFactors(i)
+            if (factors) {
+                for (const [, power] of factors) {
+                    counter += Number(power)
+                }
+            }
+
+            factorArray[i] = counter
+        }
+        return factorArray
+    }
+
+    // Create an array with the frequency of each number
+    // of factors in the corresponding index.
+    orderedFactorArray(): number[] {
+        const orderedFactorArray = []
+        const factorArray = this.factorArray()
+        for (
+            let i = 0;
+            i < natlog(this.largestValue()) / natlog(2) + 1;
+            i++
+        ) {
+            orderedFactorArray.push(0)
+        }
+
+        for (let i = 0; i < factorArray.length; i++) {
+            orderedFactorArray[factorArray[i]]++
+        }
+
+        return orderedFactorArray
+    }
+
+    // Create an array to change the bin size of the
+    // histogram to what the user asked for.
+    binFactorArray(): number[] {
+        const binFactorArray = []
+        const orderedFactorArray = this.orderedFactorArray()
+        for (
+            let i = 0;
+            i < Math.ceil(orderedFactorArray.length / this.binSize);
+            i++
+        ) {
+            binFactorArray.push(0)
+        }
+
+        let index = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            for (let j = 0; j < this.binSize; j++) {
+                if (index + j < orderedFactorArray.length) {
+                    binFactorArray[i] += orderedFactorArray[index + j]
+                }
+            }
+            index += this.binSize
+        }
+
+        return binFactorArray
+    }
+
+    // Create a number that represents how
+    // many pixels wide each bin should be.
+    binWidth(): number {
+        const binFactorArray = this.binFactorArray()
+        let binWidth = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] != 0) {
+                binWidth = i
+            }
+        }
+        return 750 / (binWidth + 1)
+    }
+
+    // Create a number that represents how many pixels high
+    // each increase of one in the bin array should be.
+    height(): number {
+        const binFactorArray = this.binFactorArray()
+        let height = binFactorArray[0]
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] > height) {
+                height = binFactorArray[i]
+            }
+        }
+        return 750 / height
+    }
+
+    draw() {
+        this.sketch.background(176, 227, 255)
```

Comment:
There are lots of magic numbers in this function. If the meaning of a constant is not clear, it needs to be commented or made symbolic. If it actually depends on the canvas size, it needs to be computed based on the canvas size.

Reply:
There is some magic left, but I put in specific comments, so resolving this one.